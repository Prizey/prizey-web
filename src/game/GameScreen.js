import React from 'react'
import { List } from 'croods'

import GoBack from 'design/GoBack/GoBack'
import Caption from 'design/Caption/Caption'
import Layout from 'design/Layout/Layout'
import Roulette from './Roulette'

const InnerComponent = ({ speed, difficulty, list }) => (
  <Layout
    leftIcon={<GoBack to="/" />}
    caption={<Caption difficulty={difficulty} />}
  >
    <Roulette
      speed={speed}
      data={list}
      onSelectItem={item =>
        // eslint-disable-next-line no-alert
        window.alert(`you tapped the item ${item.title}`)
      }
    />
  </Layout>
)

const SpeedComponent = difficulty => list => (
  <List
    name="game_settings"
    path={`/game_settings`}
    render={speedList => {
      const speed = speedList[0][`${difficulty}CarouselSpeed`]

      return InnerComponent({
        difficulty,
        list,
        speed,
      })
    }}
  />
)

export default ({ difficulty = 'easy' }) => (
  <List
    name="products"
    path={`/products/${difficulty}`}
    render={SpeedComponent(difficulty)}
  />
)
