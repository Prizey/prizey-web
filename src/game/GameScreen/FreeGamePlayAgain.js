import React from 'react'
import { connect } from 'react-redux'
import { Button, Typography } from '@material-ui/core'

import UserBalance from 'design/UserBalance'
import Layout from 'design/Layout/Layout'
import ProductImage from 'design/ProductImage'
import Caption from 'design/Caption/Caption'
import placeholder from './cardi_placeholder.png'

const Screen = ({ currentUser, location, product }) => {
  const mainText = product
    ? product.title
    : `Looks like you've already played this game, Click the "Play Again" below to Download Another App and play a new game`
  const productToShow = product || {
    image: placeholder,
  }

  return (
    <Layout
      location={location}
      currentUser={currentUser}
      leftIcon={<UserBalance />}
      confirmLeave
      caption={<Caption difficulty="easy" />}
    >
      <Typography align="center" variant="h5">
        {mainText}
      </Typography>

      <ProductImage product={productToShow} />

      <a
        href="https://www.verifyspot.net/cl.php?id=c6ab41e68cc1dd56233aa14ec2ed9859"
        target="_blank"
        rel="noopener noreferrer"
        style={{ textDecoration: 'none' }}
      >
        <Button
          variant="contained"
          style={{ backgroundColor: '#76fc03' }}
          fullWidth
        >
          Play Again
        </Button>
      </a>
    </Layout>
  )
}

export const mapState = ({ basket = {} }) => ({
  product: basket.product,
})

export default connect(mapState)(Screen)
