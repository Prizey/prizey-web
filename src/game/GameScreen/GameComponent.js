import React from 'react'
import { connect } from 'react-redux'
import { Typography } from '@material-ui/core'

import { chooseProduct } from 'store/basket/actions'

import UserBalance from 'design/UserBalance'
import Caption from 'design/Caption/Caption'
import Layout from 'design/Layout/Layout'
import ProfileLink from 'design/ProfileLink/ProfileLink'
import Roulette from '../Roulette'

export default connect(
  null,
  { chooseProduct },
)(
  ({
    speed,
    difficulty,
    list,
    multiplier,
    currentUser,
    location,
    freegame,
    hideLogin,
    ...props
  }) => (
    <Layout
      leftIcon={<UserBalance />}
      caption={<Caption difficulty={difficulty} />}
      location={location}
      currentUser={currentUser}
      rightIcon={!hideLogin && <ProfileLink confirmLeave />}
      confirmLeave
    >
      <Roulette
        aria-label="roulette"
        speed={speed}
        data={list}
        multiplier={multiplier}
        onSelectItem={item => {
          props.chooseProduct(item)
          if (freegame) {
            props.navigate(`/freegame-play-again`)
          } else {
            props.navigate(`/game/${difficulty}/claim`)
          }
        }}
      />
      <br />
      <Typography align="center" variant="body2" color="textSecondary">
        Tap the screen, <br />
        win what you tap.
      </Typography>
    </Layout>
  ),
)
