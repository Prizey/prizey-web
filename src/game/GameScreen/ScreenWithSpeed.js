import React from 'react'
import SpeedComponent from '../SpeedComponent'
import GameComponent from './GameComponent'

export default (difficulty, navigate, currentUser, location) => list => (
  <SpeedComponent
    render={settings => {
      const speed = settings[`${difficulty}CarouselSpeed`]

      return (
        <GameComponent
          location={location}
          currentUser={currentUser}
          list={list}
          difficulty={difficulty}
          multiplier={settings.priceMultiplier}
          navigate={navigate}
          speed={speed}
        />
      )
    }}
  />
)
