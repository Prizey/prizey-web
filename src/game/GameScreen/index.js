import React from 'react'
import { List, Info } from 'croods'
import { Redirect } from '@reach/router'
import ScreenWithSpeed from './ScreenWithSpeed'

export default ({
  freegame,
  currentUser,
  location,
  navigate,
  difficulty = 'easy',
  hideLogin,
}) => {
  if (freegame) {
    return (
      <Info
        id={1}
        disableCache
        name="freegameIps"
        path="/freegame_ips"
        method="POST"
        render={info => {
          if (info && info.ipBlocked) {
            navigate('/freegame-play-again')
          }

          return (
            <List
              disableCache
              name="products"
              path={`/products/${difficulty}`}
              render={ScreenWithSpeed(
                difficulty,
                navigate,
                currentUser,
                location,
                freegame,
                hideLogin,
              )}
            />
          )
        }}
      />
    )
  }

  return currentUser ? (
    <List
      name="products"
      path={`/products/${difficulty}`}
      render={ScreenWithSpeed(difficulty, navigate, currentUser, location)}
    />
  ) : (
    <Redirect to={`/sign-in?next=/game/${difficulty}`} noThrow />
  )
}
