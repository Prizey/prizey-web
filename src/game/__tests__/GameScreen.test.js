import React from 'react'
import renderer from 'react-test-renderer'
import GameScreen from '../GameScreen'

const mockList = [
  { id: 1, image: '/mocks/trump-mask.png', title: 'trump' },
  { id: 2, image: '/mocks/sweatshirt.png', title: 'sweatshirt' },
  { id: 3, image: '/mocks/shoe.png', title: 'shoe' },
]

jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component {...props} chooseProduct={jest.fn} />
  ),
}))

jest.mock('croods', () => ({
  Info: ({ children, ...props }) => (
    <div {...props}>
      Info -{' '}
      {props.render({
        easyCarouselSpeed: 500,
        priceMultiplier: 1,
      })}
    </div>
  ),
  List: props => (
    <div {...props}>
      {props.children} - Render: {props.render(mockList)}
    </div>
  ),
  Provider: ({ children, ...props }) => (
    <div {...props}>Provider - {children}</div>
  ),
  createReducer: () => (state = {}) => state,
}))

it('renders correctly', () => {
  const tree = renderer.create(<GameScreen currentUser />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('trigger the alert', () => {
  const navigate = jest.fn()
  const tree = renderer.create(<GameScreen currentUser navigate={navigate} />)
    .root

  tree
    .findByProps({ 'aria-label': 'roulette' })
    .props.onSelectItem({ title: 'item' })
  expect(navigate).toHaveBeenCalledTimes(1)
})

describe("when we don't have the currentUser yet", () => {
  it('redirects correctly', () => {
    const tree = renderer.create(<GameScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
