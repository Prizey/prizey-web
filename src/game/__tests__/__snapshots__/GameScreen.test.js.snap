// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  name="products"
  path="/products/easy"
  render={[Function]}
>
   - Render: 
  <div
    id={1}
    name="game"
    path="/game_setting"
    render={[Function]}
  >
    Info -
     
    <div
      className="Component-root-8"
    >
      <div
        className="Component-header-6"
      >
        <a
          aria-label="My profile"
          className="MuiButtonBase-root-16 MuiIconButton-root-10 Component-root-9"
          href="/profile"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiIconButton-label-15"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-19 MuiSvgIcon-fontSizeLarge-27"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <g>
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83z"
                />
                <path
                  d="M12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </g>
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-71"
          />
        </a>
      </div>
      <div
        className="Component-container-3"
      >
        <img
          alt="Prizey"
          className="Component-logoImage-7"
          src="/logo.png"
        />
        <div
          className="Component-caption-1"
        >
          <div
            className="Component-root-32"
          >
            <span
              className="Component-icon-28 Component-icon_easy-29"
            />
            <h6
              className="MuiTypography-root-33 MuiTypography-subtitle1-51"
            >
              EASY
            </h6>
          </div>
        </div>
        <div
          className="Component-children-2"
        >
          <div
            className="Component-root-70"
          >
            <div
              aria-label="rouletteItem-0"
              onMouseDown={[Function]}
            >
              <img
                alt="trump"
                className="Component-img-69"
                pose="enter"
                speed={NaN}
                src="/mocks/trump-mask.png"
              />
            </div>
            <div
              aria-label="rouletteItem-1"
              onMouseDown={[Function]}
            >
              <img
                alt="sweatshirt"
                className="Component-img-69"
                pose="leave"
                speed={NaN}
                src="/mocks/sweatshirt.png"
              />
            </div>
            <div
              aria-label="rouletteItem-2"
              onMouseDown={[Function]}
            >
              <img
                alt="shoe"
                className="Component-img-69"
                pose="leave"
                speed={NaN}
                src="/mocks/shoe.png"
              />
            </div>
          </div>
          <br />
          <p
            className="MuiTypography-root-33 MuiTypography-body2-41 MuiTypography-colorTextSecondary-66 MuiTypography-alignCenter-56"
          >
            Tap the screen, 
            <br />
            win what you tap.
          </p>
        </div>
        <div
          className="Component-footer-4"
        >
          <a
            className="Component-footerLink-5"
            href="https://app.termly.io/document/terms-and-conditions/4de5f3bb-a333-47bb-9daf-f82f7d4e1cfb"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-33 MuiTypography-caption-43 MuiTypography-colorTextSecondary-66 MuiTypography-alignCenter-56"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-33 MuiTypography-caption-43 MuiTypography-colorTextSecondary-66 MuiTypography-alignCenter-56"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-33 MuiTypography-caption-43 MuiTypography-colorTextSecondary-66 MuiTypography-alignCenter-56"
            >
              FAIRNESS
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-33 MuiTypography-caption-43 MuiTypography-colorTextSecondary-66 MuiTypography-alignCenter-56"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when we don't have the currentUser yet redirects correctly 1`] = `null`;
